sympy/liealgebras/type_e.py

```diff
@@ -47,228 +47,62 @@ def simple_root(self, i):
-                                        if d == 1:
-                                            root[3] = Rational(-1, 2)
-                                        if e == 1:
-                                            root[4] = Rational(-1, 2)
-                                        if f == 1:
-                                            root[5] = Rational(1, 2)
-                                        posroots[k] = root
-
-            return posroots
-        if n == 8:
-            posroots = {}
-            k = 0
-            for i in range(n):
-                for j in range(i+1, n):
-                    k += 1
-                    root = self.basic_root(i, j)
-                    posroots[k] = root
-                    k += 1
-                    root = self.basic_root(i, j)
-                    root[i] = 1
-                    posroots[k] = root
-
-            root = [Rational(1, 2), Rational(1, 2), Rational(1, 2), Rational(1, 2), Rational(1, 2),
-                    Rational(-1, 2), Rational(-1, 2), Rational(1, 2)]
-            for a in range(0, 2):
-                for b in range(0, 2):
-                    for c in range(0, 2):
-                        for d in range(0, 2):
-                            for e in range(0, 2):
-                                for f in range(0, 2):
-                                    for g in range(0, 2):
-                                        if (a + b + c + d + e + f + g)%2 == 0:
-                                            k += 1
-                                            if a == 1:
-                                                root[0] = Rational(-1, 2)
-                                            if b == 1:
-                                                root[1] = Rational(-1, 2)
-                                            if c == 1:
-                                                root[2] = Rational(-1, 2)
-                                            if d == 1:
-                                                root[3] = Rational(-1, 2)
-                                            if e == 1:
-                                                root[4] = Rational(-1, 2)
-                                            if f == 1:
-                                                root[5] = Rational(1, 2)
-                                            if g == 1:
-                                                root[6] = Rational(1, 2)
-                                            posroots[k] = root
-
-            return posroots
+        n = self.rank
+        e = eye(8)
+
+        if i == n:
+            root =  e.row(n-3) - e.row(n-2)
+        elif i == n - 1:
+            root = - ones(1,8) / 2
+        elif i == n - 2:
+            root = e.row(n-3) + e.row(n-2)
+        else:
+            root = e.row(i-1) - e.row(i)
+        root = root.tolist()[0]
+        if n != 8:
+            root[n-1] = sqrt(sum(root[j]**2 for j in range(n-1, 8)))
```

Comment:
I would use the precomputed `sqrt(9-n)/2` instead of the runtime norm computation.

Reply:
I changed it and left a comment. You need to multiply by value at the last index of the E8 simple root since its a sum. If it is zero, it will remain zero, otherwise your suggestion will work.